File: Classes/GameMap/FightPoint.py
class FightPoint:
    def __init__(self, point, allowedDirections, boolean = True):
        """
        :param point: (float, float)
        :param allowedDirections: list of allowed directions (each direction receives "UP", "DOWN", "LEFT" or "RIGHT")
        :param boolean: indicates if the point is activated
        """
        self._point = point
        self._allowedDirections = allowedDirections
        self._activated = boolean

    def activate(self):
        self.setActivated(True)

    def deactivate(self):
        self.setActivated(False)

    def getPoint(self):
        return self._point

    def setPoint(self, point):
        self._point = point

    def getAllowedDirections(self):
        return self._allowedDirections

    def setAllowedDirections(self, allowedDirecions):
        self._allowedDirections = allowedDirecions

    def getActivated(self):
        return self._activated

    def setActivated(self, boolean):
        self._activated = boolean

File: TheFlameRealmJava/core/src/main/java/com/flamerealm/game/gamemap/FightPoint.java
package com.flamerealm.game.gamemap;

import com.badlogic.gdx.math.Vector2;
import com.flamerealm.game.assets.EncounterManifest;

import java.util.List;

/**
 * Port de FightPoint.py. O manifest diz qual EncounterManifest carregar ao
 * ativar este ponto (null para pontos que nao disparam combate, ex.: startPoint).
 */
public class FightPoint {
    private Vector2 point;
    private List<String> allowedDirections;
    private boolean activated;
    private EncounterManifest manifest;

    public FightPoint(Vector2 point, List<String> allowedDirections) {
        this(point, allowedDirections, true, null);
    }

    public FightPoint(Vector2 point, List<String> allowedDirections, boolean activated) {
        this(point, allowedDirections, activated, null);
    }

    public FightPoint(Vector2 point, List<String> allowedDirections, boolean activated, EncounterManifest manifest) {
        this.point = point;
        this.allowedDirections = allowedDirections;
        this.activated = activated;
        this.manifest = manifest;
    }

    public void activate() {
        setActivated(true);
    }

    public void deactivate() {
        setActivated(false);
    }

    public Vector2 getPoint() {
        return point;
    }

    public void setPoint(Vector2 point) {
        this.point = point;
    }

    public List<String> getAllowedDirections() {
        return allowedDirections;
    }

    public void setAllowedDirections(List<String> allowedDirections) {
        this.allowedDirections = allowedDirections;
    }

    public boolean getActivated() {
        return activated;
    }

    public void setActivated(boolean activated) {
        this.activated = activated;
    }

    public EncounterManifest getManifest() {
        return manifest;
    }

    public void setManifest(EncounterManifest manifest) {
        this.manifest = manifest;
    }
}
